server/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -107,13 +108,13 @@
     final ShardsAllocator shardsAllocator;
 
     public ClusterModule(Settings settings, ClusterService clusterService, List<ClusterPlugin> clusterPlugins,
-                         ClusterInfoService clusterInfoService) {
+                         ClusterInfoService clusterInfoService, ThreadPool threadPool) {
```

Comment:
since we only need the `ThreadContext` here can we pass that as the constructor argument instead of the `ThreadPool`?